packages/lesswrong/lib/collections/users/helpers.ts

```diff
@@ -306,47 +307,84 @@ export const userGetLocation = (currentUser: UsersCurrent|DbUser|null): {
   return {lat: placeholderLat, lng: placeholderLng, known: false}
 }
 
-// Return the current user's location, as a latitude-longitude pair, plus
-// boolean fields `loading` and `known`. If `known` is false, the lat/lng are
-// invalid placeholders. If `loading` is true, then `known` is false, but the
-// state might be updated with a location later.
-//
-// If the user is logged in, the location specified in their account settings
-// is used first. If the user is not logged in, then no location is available
-// for server-side rendering, but we can try to get a location client-side
-// using the browser geolocation API. (This won't necessarily work, since not
-// all browsers and devices support it, and it requires user permission.)
-export const useUserLocation = (currentUser: UsersCurrent|DbUser|null) => {
+/**
+ * Return the current user's location, by checking a few places.
+ *
+ * If the user is logged in, the location specified in their account settings is used first.
+ * If the user is not logged in, then no location is available for server-side rendering,
+ * but we can check if we've already saved a location in their browser's local storage.
+ *
+ * If we've failed to get a location for the user, finally try to get a location
+ * client-side using the browser geolocation API.
+ * (This won't necessarily work, since not all browsers and devices support it, and it requires user permission.)
+ * This step is skipped if the "dontAsk" flag is set, to be less disruptive to the user
+ * (for example, on the forum homepage).
+ *
+ * @param {UsersCurrent|DbUser|null} currentUser - The user we are checking.
+ * @param {boolean} dontAsk - Flag that prevents us from asking the user for their browser's location.
+ *
+ * @returns {Object} location
+ * @returns {number} location.lat - The user's latitude.
+ * @returns {number} location.lng - The user's longitude.
+ * @returns {boolean} location.loading - Indicates that we might have a known location later.
+ * @returns {boolean} location.known - If false, then we're returning the default location instead of the user's location.
+ * @returns {string} location.label - The string description of the location (ex: Cambridge, MA, USA).
+ * @returns {Function} location.setLocation - Function to set the location directly.
+ */
+export const useUserLocation = (currentUser: UsersCurrent|DbUser|null, dontAsk?: boolean): {
```

Comment:
This is quite large now, maybe I should move it somewhere else? Not sure where

Reply:
I'm mildly ok with this.